fdbserver/Status.actor.cpp

```diff
@@ -1024,19 +1024,38 @@ static JsonBuilderObject clientStatusFetcher(std::map<NetworkAddress, std::pair<
 	return clientStatus;
 }
 
-ACTOR static Future<JsonBuilderObject> recoveryStateStatusFetcher(WorkerDetails mWorker, int workerCount, std::set<std::string> *incomplete_reasons, int* statusCode) {
+ACTOR static Future<JsonBuilderObject> recoveryStateStatusFetcher(Database cx, WorkerDetails mWorker, int workerCount, std::set<std::string> *incomplete_reasons, int* statusCode) {
 	state JsonBuilderObject message;
-
+	state Transaction tr(cx);
 	try {
-		state Future<TraceEventFields> activeGens = timeoutError(mWorker.interf.eventLogRequest.getReply( EventLogRequest( LiteralStringRef("MasterRecoveryGenerations") ) ), 1.0);
-		TraceEventFields md = wait( timeoutError(mWorker.interf.eventLogRequest.getReply( EventLogRequest( LiteralStringRef("MasterRecoveryState") ) ), 1.0) );
+		state Future<TraceEventFields> mdActiveGensF = timeoutError(mWorker.interf.eventLogRequest.getReply( EventLogRequest( LiteralStringRef("MasterRecoveryGenerations") ) ), 1.0);
+		state Future<TraceEventFields> mdF = timeoutError(mWorker.interf.eventLogRequest.getReply( EventLogRequest( LiteralStringRef("MasterRecoveryState") ) ), 1.0);
+		state Future<TraceEventFields> mDBAvailableF = timeoutError(mWorker.interf.eventLogRequest.getReply( EventLogRequest( LiteralStringRef("MasterRecoverFDBAvailable") ) ), 1.0);
+		state Future<Version> rvF = timeoutError(tr.getReadVersion(), 1.0);
```

Comment:
Hmm, I guess one downside of relying on getting a read version to do this computation is that we will fail if the database is unavailable or taking longer than 1 second to hand them out. We could mitigate this by using immediate priority, and we should probably also use `errorOr` to prevent us from failing this whole block in that event. I guess our best option in the case of an error would be to omit the time since recovered message.

Reply:
Done.